### PR TITLE
Misc improvements

### PR DESCRIPTION
**Pull request by allanlewis**
- Allow package to be installed in editable mode
- elements.Null.to_yaml: Correct return type
- elements.Workflow: Allow nulls as `on` values
- elements: Add reprs
- elements.Job: Allow configuring job name
- elements: Replace 'star import' with explicit imports
- Define module exports

### CODE REVIEW

**Review by joshaw — APPROVED** [review]

File: gadk/elements.py [hunk 1]
```diff
@@ -111,6 +141,14 @@ def __init__(
         self._action = action
         self._with: Dict[str, str] = with_args or {}
 
+    def __repr__(self):
+        slug = (
+            f"id={self._id!r}" if self._id is not None
```
Comment (afshelley) [inline gadk/elements.py:146]: Why use `!r` rather than `=` here as above?

Comment (allanlewis) [reply gadk/elements.py:146]: Good spot - I'll change that.

File: gadk/elements.py [hunk 2]
```diff
@@ -162,11 +202,20 @@ def __init__(
         if default_checkout:
             self._steps.insert(0, UsesStep(action=ACTION_CHECKOUT))
 
+    def __repr__(self):
+        repr_ = f"<{type(self).__name__}"
+        if self._name is not None:
+            repr_ = f"{repr_} {self._name=}"
+
+        return f"{repr_} {self._steps=}>"
```
Comment (afshelley) [inline gadk/elements.py:210]: !notimportant - I found this a bit difficult to parse, could it be simpler? e.g.
```suggestion
        name_str = f"{self._name=} " if self._name is not None else "" 
        return f"<{type(self).__name__} {name_str}{self._steps=}>"
```

**Review by afshelley — APPROVED** [review]
Some very small comments but all looks fine, happy for you to merge

File: gadk/elements.py (hunk 2 above)
Comment (allanlewis) [reply gadk/elements.py:210]: I think I marginally prefer my approach here - fixed preamble, optional bit in the middle, fixed postamble.

**Comment by allanlewis** [comment]
Oops, I fixed a thing then merged before pushing 🤦

**Comment by allanlewis** [comment]
Sorry for the mess: #7.

**Pull request merged**